File: web/src/test/java/top/bettercode/summer/web/WebTestController.java
package top.bettercode.summer.web;


import java.util.Date;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpStatus;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import top.bettercode.summer.tools.lang.property.Settings;
import top.bettercode.summer.tools.lang.util.StringUtil;
import top.bettercode.summer.web.exception.BusinessException;
import top.bettercode.summer.web.form.FormDuplicateCheck;
import top.bettercode.summer.web.resolver.Cent;
import top.bettercode.summer.web.resolver.CentConverter;
import top.bettercode.summer.web.support.code.CodeService;
import top.bettercode.summer.web.support.code.ICodeService;
import top.bettercode.summer.web.validator.ChinaCell;

/**
 * @author Peter Wu
 */
@RestController
@Validated
public class WebTestController extends BaseController {

  @FormDuplicateCheck
  @RequestMapping(value = "/webtest")
  public Object test(@Validated DataDicBean form, @Cent Long cent, Date a, @ChinaCell String cell) {
    System.err.println(a);
    System.err.println(cent);
    System.err.println(form.getPrice());
    System.err.println(StringUtil.valueOf(form, true));
    DataDicBean dataDicBean = new DataDicBean();
    dataDicBean.setCode("code");
    dataDicBean.setIntCode(1);
    dataDicBean.setPath("/abc.jpg");
    return ok(dataDicBean);
  }

  @FormDuplicateCheck
  @RequestMapping(value = "/weberrors")
  public Object error() {
    throw new BusinessException(String.valueOf(HttpStatus.BAD_GATEWAY.value()), "xx");
  }

  @Configuration(proxyBeanMethods = false)
  protected static class CodeConfiguration {

    @Bean
    public CentConverter yuanToCentConverter() {
      return new CentConverter();
    }

    @Bean
    public ICodeService codeNumberService() {
      return new CodeService(Settings.getDicCode());
    }


  }
}

File: web/src/test/kotlin/top/bettercode/summer/test/web/WebTestController.kt
package top.bettercode.summer.test.web

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpStatus
import org.springframework.util.Assert
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import top.bettercode.summer.tools.lang.property.Settings.dicCode
import top.bettercode.summer.tools.lang.util.StringUtil.json
import top.bettercode.summer.web.BaseController
import top.bettercode.summer.web.exception.SystemException
import top.bettercode.summer.web.form.FormDuplicateCheck
import top.bettercode.summer.web.resolver.Unit
import top.bettercode.summer.web.support.code.CodeService
import top.bettercode.summer.web.support.code.ICodeService
import top.bettercode.summer.web.validator.ChinaCell
import java.util.*

/**
 * @author Peter Wu
 */
@RestController
@Validated
class WebTestController : BaseController() {
    @FormDuplicateCheck
    @RequestMapping(value = ["/webtest"])
    fun test(@Validated form: DataDicBean, @Unit cent: Long?, a: Date?, @ChinaCell cell: String?): Any {
        System.err.println(a)
        System.err.println(cent)
        Assert.isTrue(cent == 2200L, "cent != 2200")
        System.err.println(form.price)
        System.err.println(json(form, true))
        val dataDicBean = DataDicBean()
        dataDicBean.code = "code"
        dataDicBean.intCode = 1
        dataDicBean.path = "/abc.jpg"
        return ok(dataDicBean)
    }

    @FormDuplicateCheck
    @RequestMapping(value = ["/weberrors"])
    fun error(): Any {
        throw SystemException(HttpStatus.BAD_GATEWAY.value().toString(), "xx")
    }

    @Configuration(proxyBeanMethods = false)
    protected class CodeConfiguration {
        @Bean
        fun codeService(): ICodeService {
            return CodeService(dicCode)
        }
    }
}